From 8b5d05772077b14c7845781bbf4ca46560199207 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Tue, 21 Jul 2026 19:23:49 -0400
Subject: [PATCH 8/8] AOSCOS: Guard stray inclusion of Simulator-riscv64.h

This is a leftover by Bug 2038968 D300034, but it should be fixed in Bug
2039343 D300387 which happens to miss the Fx152 train.

Should be dropped by the release of Fx153.

Link: https://phabricator.services.mozilla.com/D300034 ("Bug 2038968 -
 Part 4: Make it an error to include simulator headers when simulator is
 disabled. r=jandem!")
Link: https://phabricator.services.mozilla.com/D300387 ("Bug 2039343 -
 Part 6: Remove InstructionGetters. r=jandem!")

Co-Authored-By: Rong Bao <webmaster@csmantle.top>
---
 .../constant/Base-constant-riscv-cpp.patch      | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch

diff --git a/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch b/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch
new file mode 100644
index 0000000000000..987693b5412c5
--- /dev/null
+++ b/src/js/src/jit/riscv64/constant/Base-constant-riscv-cpp.patch
@@ -0,0 +1,17 @@
+diff --git a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
+index b25ce174bc5bbb3a01ebd7676c91afa7b3d5a628..40b7472ccec197c88de7c7a66809b8822038dc6c 100644
+--- a/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
++++ b/js/src/jit/riscv64/constant/Base-constant-riscv.cpp
+@@ -13,7 +13,11 @@
+ #include "jit/riscv64/constant/Constant-riscv-v.h"
+ #include "jit/riscv64/constant/Constant-riscv-zicsr.h"
+ #include "jit/riscv64/constant/Constant-riscv-zifencei.h"
+-#include "jit/riscv64/Simulator-riscv64.h"
++
++#ifdef JS_SIMULATOR_RISCV64
++#  include "jit/riscv64/Simulator-riscv64.h"
++#endif
++
+ namespace js {
+ namespace jit {
+ 
-- 
2.52.0

